File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        p1 = 0
        p2 = 1

        while (p1 <= len(nums) - 1):
            if((nums[p1] + nums[p2]) == target):
                return [p1, p2]
            elif(p2 == (len(nums) - 1)):
                p1 += 1
                p2 = p1 + 1
            else:
                p2 += 1
        

File: Data Structures & Algorithms/binary-search/submission-0.cs
public class Solution {
    public int Search(int[] nums, int target) {
        int l = 0;
        int r = nums.Length - 1;
        int p = (r + l) / 2;

        while (l <= r) {
            if (nums[p] == target) {
                return p;
            }
            else if (nums[p] < target) {
                l = p + 1;
            }
            else if (nums[p] > target) {
                r = p - 1;
            }
            p = (r + l) / 2;
        }

        return -1;
    }
}
